Core/include/Acts/Propagator/MultiStepperAborters.hpp

```diff
@@ -99,14 +70,23 @@ struct MultiStepperSurfaceReached {
       }
     }
 
-    // These values are changed by the single component aborters but must be
-    // reset if not all components are on the target
-    if (!reached) {
-      navigator.currentSurface(state.navigation, oldCurrentSurface);
-      navigator.targetReached(state.navigation, false);
+    bool reached = true;
+
+    for (auto cmp : stepper.componentIterable(state.stepping)) {
+      // note that this is not copying anything heavy
+      auto singleState = cmp.singleState(state);
+      const auto& singleStepper = cmp.singleStepper(stepper);
+
+      if (!SurfaceReached::operator()(singleState, singleStepper, navigator,
+                                      logger)) {
+        reached = false;
+      } else {
+        cmp.status() = Acts::Intersection3D::Status::onSurface;
+      }
     }
 
     return reached;
```

Comment:
Maybe we should add VERBOSE log messages like above for this case? (like the "maybe next time?" log message below)

Reply:
Ah actually I now see that we get a bunch of logging messages when calling the parent `operator()`...

But I thought a closing VERBOSE message reporting the status of `reached` wouldn't hurt right?